CRM/Utils/System/Drupal8.php

```diff
@@ -634,6 +634,16 @@ public function synchronizeUsers() {
     );
   }
 
+  /**
+   * @inheritDoc
+   */
+  public function setMessage($message) {
+    // CiviCRM sometimes includes markup in messages (ex: Event Cart)
+    // it needs to be rendered before being displayed.
+    $message = \Drupal\Core\Render\Markup::create($message);
```

Comment:
This isn't really "rendering" it - this is saying that this markup is safe and requires no filtering (otherwise Twig autoescaping will take effect later). Is this what we want here? Or, should some kind of filtering be applied?

Reply:
Hmm, good point. At worst, I think there could be admin-specified input, for example the Event Title, but presumably those things would/should be filtered out before being sent to `setMessage`.

For example:

```
    $url = CRM_Utils_System::url('civicrm/event/view_cart');
    CRM_Utils_System::setUFMessage(ts("<b>%1</b> has been added to your cart. <a href='%2'>View your cart.</a>", array(
          1 => $event_in_cart->event->title,
          2 => $url,
        )));
```

So the `$event_in_cart->event->title` could be malicious.

Although in this specific case, CiviCRM is filtering-on-input:

![Capture d’écran de 2019-04-04 20-43-19](https://user-images.githubusercontent.com/254741/55597190-4cc11800-571a-11e9-9c19-0862808cfbe4.png)

I think we should filter-on-output most of the time, and we are probably pretty bad at doing so, but if we did so, it should probably be in the `ts` function?

Reply:
Ping @dsnopek ?